lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -374,7 +374,8 @@ namespace {
           // Optimization: If the constructor heads don't match, subspace is
           // impossible.
 
-          if (this->Head != other.Head) {
+          if (this->Head.getBaseIdentifier() !=
```

Comment:
This seems like it was leftover after your rebase.

Reply:
This was actually a change so that all the tests would pass as intended.  Specifically in cases like this:
```
enum PatternCasts {
  case one(Any)
  case two
  case three(String)
}

func checkPatternCasts() {
  // Pattern casts with this structure shouldn't warn about duplicate cases.
  let x: PatternCasts = .one("One")
  switch x {
  case .one(let s as String): print(s)
  case .one: break
  case .two: break
  case .three: break
  }

  // But should warn here.
  switch x {
  case .one(_): print(s)
  case .one: break // expected-warning {{case is already handled by previous patterns; consider removing it}}
  case .two: break
  case .three: break
  }
}
```
Just because the whole DeclName doesn't match doesn't necessarily mean that a subspace is impossible, only if the base is different does it mean that it is impossible.  If we test against the whole DeclName the second switch statement does not warn as expected.

Reply:
Ugh, this is actually because we rewrite the pattern but not the name associated with the pattern.  This is fine, then.